drivers/ieee802154/ieee802154_rf2xx_iface.c

```diff
@@ -0,0 +1,368 @@
+/* ieee802154_rf2xx_iface.c - ATMEL RF2XX IEEE 802.15.4 Interface */
+
+/*
+ * Copyright (c) 2019 Gerson Fernando Budke
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME ieee802154_rf2xx_iface
+#define LOG_LEVEL CONFIG_IEEE802154_DRIVER_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <errno.h>
+#include <assert.h>
+
+#include <device.h>
+#include <drivers/spi.h>
+#include <drivers/gpio.h>
+
+#include "ieee802154_rf2xx.h"
+#include "ieee802154_rf2xx_regs.h"
+#include "ieee802154_rf2xx_iface.h"
+
+void rf2xx_iface_phy_rst(struct device *dev)
+{
+	const struct rf2xx_config *conf = dev->config->config_info;
+	struct rf2xx_context *ctx = dev->driver_data;
+
+	/* Ensure control lines have correct levels. */
+	gpio_pin_write(ctx->reset_gpio, conf->reset.pin, 1);
+	gpio_pin_write(ctx->slptr_gpio, conf->slptr.pin, 0);
+
+	/* Wait typical time of timer TR1. */
+	k_busy_wait(330);
+
+	gpio_pin_write(ctx->reset_gpio, conf->reset.pin, 0);
+	k_busy_wait(10);
+	gpio_pin_write(ctx->reset_gpio, conf->reset.pin, 1);
+}
+void rf2xx_iface_phy_tx_start(struct device *dev)
+{
+	const struct rf2xx_config *conf = dev->config->config_info;
+	struct rf2xx_context *ctx = dev->driver_data;
+
+	/* Start TX transmission at rise edge */
+	gpio_pin_write(ctx->slptr_gpio, conf->slptr.pin, 1);
+	/* 16.125[μs] delay to detect signal */
+	k_busy_wait(20);
+	/* restore initial pin state */
+	gpio_pin_write(ctx->slptr_gpio, conf->slptr.pin, 0);
+}
+
+u8_t rf2xx_iface_reg_read(struct device *dev, u8_t addr)
+{
+	struct rf2xx_context *ctx = dev->driver_data;
+	u8_t regval = 0;
+
+	addr |= RF_CMD_REG_R;
+
+	const struct spi_buf tx_buf = {
+		.buf = &addr,
+		.len = 1
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+	const struct spi_buf rx_buf = {
+		.buf = &regval,
+		.len = 1
+	};
+	const struct spi_buf_set rx = {
+		.buffers = &rx_buf,
+		.count = 1
+	};
+
+	if (spi_transceive(ctx->spi, &ctx->spi_cfg, &tx, NULL) != 0) {
+		LOG_ERR("Failed to exec rf2xx_reg_read CMD at address %d",
+			addr);
+	}
+	if (spi_transceive(ctx->spi, &ctx->spi_cfg, NULL, &rx) != 0) {
+		LOG_ERR("Failed to exec rf2xx_reg_read READ at address %d",
+			addr);
+	}
+	spi_release(ctx->spi, &ctx->spi_cfg);
```

Comment:
I saw you are using SPI_LOCK_ON, so really you hw need first a transaction to tell the register you'll read, then another one for the actual response of that command?

It cannot be done at once during the same transaction? 
It really needs the CS to be put off, and on again between the register request, and the actual data ?

Sounds really weird. Please reference that from actual hw datasheet if that's the case.

(same comment applies to all the double spi_transceive() below)

Reply:
I faced problems when start using spi driver and I try to keep simple to test. However I need to use the options that maintain CS low and lock to avoid access to spi from another software.

After review I improve spi transactions with only one access. This need more tests to garantee that everything is working properly.